cli/module_generate/scripts/generate_stubs.py

```diff
@@ -71,6 +83,37 @@ def main(
                 )
                 i = f"from {stmt.module} import {i_strings}"
                 imports.append(i)
+            elif isinstance(stmt, ast.ClassDef) and stmt.name == resource_name:
+                for cstmt in stmt.body:
+                    if isinstance(cstmt, ast.ClassDef):
+                        nodes.append(cstmt.name)
+                    elif isinstance(cstmt, ast.AnnAssign):
+                        nodes.append(cstmt.target.id)
+                    elif isinstance(cstmt, ast.AsyncFunctionDef):
+                        for arg in cstmt.args.args:
+                            if isinstance(arg.annotation, ast.Name) and arg.annotation.id in nodes:
+                                arg.annotation = return_attribute(resource_name, arg.annotation.id)
+                            elif isinstance(arg.annotation, ast.Subscript):
+                                if isinstance(arg.annotation.slice, ast.Subscript):
+                                    if isinstance(arg.annotation.slice.slice, ast.Name) and arg.annotation.slice.slice.id in nodes:
+                                        arg.annotation.slice.slice = return_attribute(resource_name, arg.annotation.slice.slice.id)
```

Comment:
We have some instances where we have `Subscript`'s in `Subscript`'s (i.e. `Dict[str, List[str]]`). Since that exists in our Python SDK, I wanted to protect us if it ever happened with a custom class. There is however, the unfortunate truth that if someone decides to put a custom class nested a third time, it will fail. But thankfully this test will fail if that happens.

Reply:
This is a good use case for recursion